airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -3247,6 +3251,35 @@ def _format_schema_for_description(schema: dict) -> list:
     return description
 
 
+class ImpersonationToken:
+    """Simulate the interface of gcloud.aio.auth.token.BaseToken and generate impersonation_chain access_token"""
+
+    def __init__(self, project_id: str | None, impersonation_account: str) -> None:
+        self.project_id = project_id
+        self.impersonation_account = impersonation_account
+
+    async def get_project(self) -> str | None:
+        project = (
+            self.project_id
+            or os.environ.get("GOOGLE_CLOUD_PROJECT")
+            or os.environ.get("GCLOUD_PROJECT")
+            or os.environ.get("APPLICATION_ID")
+        )
+        return project
+
+    async def get(self) -> str | None:
+        creds, _ = google.auth.default()
+
+        impersonated_creds = impersonated_credentials.Credentials(
+            source_credentials=creds,
+            target_principal=self.impersonation_account,
```

Comment:
`impersonation_chain` can have multiple service accounts in the chain. you'll have to handle that by doing something similar to `_get_target_principal_and_delegates: https://github.com/apache/airflow/blob/e2393ee6dd3a927e753e6375621af07aa0c734dc/airflow/providers/google/common/hooks/base_google.py#L273-L285

Reply:
Thanks for reminding me! just addressed